tests/Unirest/RequestTest.php

```diff
@@ -83,6 +83,33 @@ public function testDefaultHeader()
         $this->assertFalse(property_exists($response->body->headers, 'hello'));
     }
 
+    public function testConnectionReuse()
+    {
+        for ($i = 0; $i < 2; $i++) {
+            $url = "http://httpbin.org/get";
+
+            $_headers = [
+                'Accept' => 'application/json'
+            ];
+
+            Request::get($url, $_headers);
+
+            sleep(1);
+        }
+
+        $this->assertEquals(1, Request::$prevCallSuccessfulConnects);
```

Comment:
Instead of accessing the public property like this, make it protected then add a child class to access the parent's property using a public getter in the child class, Just like you did in JsonMapper's test case